miner/worker.go

```diff
@@ -549,38 +515,47 @@ func (w *worker) taskLoop() {
 func (w *worker) resultLoop() {
 	for {
 		select {
-		case result := <-w.resultCh:
+		case block := <-w.resultCh:
 			// Short circuit when receiving empty result.
-			if result == nil {
+			if block == nil {
 				continue
 			}
 			// Short circuit when receiving duplicate result caused by resubmitting.
-			block := result.block
 			if w.chain.HasBlock(block.Hash(), block.NumberU64()) {
 				continue
 			}
-			// Update the block hash in all logs since it is now available and not when the
-			// receipt/log of individual transactions were created.
-			for _, r := range result.receipts {
-				for _, l := range r.Logs {
-					l.BlockHash = block.Hash()
-				}
+			sealhash := w.engine.SealHash(block.Header())
```

Comment:
Could we replace this with

```go
var (
    sealhash = w.engine.SealHash(block.Header())
    hash     = block.Hash()
)
```

and then use these below? It would prevent requesting the hash from the block for every single log.